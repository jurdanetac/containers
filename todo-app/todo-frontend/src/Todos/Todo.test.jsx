import renderer from "react-test-renderer";
import Todo from "./Todo";

it("displays correct information", () => {
  const todo = {
    text: "This is a todo",
    done: false,
  };

  const mockDeleteTodo = jest.fn();
  const mockCompleteTodo = jest.fn();

  const component = renderer.create(
    <Todo
      todo={todo}
      deleteTodo={mockDeleteTodo}
      completeTodo={mockCompleteTodo}
    />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(tree.children[0].children[0]).toBe(todo.text);
});
