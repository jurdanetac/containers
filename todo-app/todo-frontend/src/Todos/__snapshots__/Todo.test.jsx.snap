// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays correct information 1`] = `
<div
  style={
    {
      "display": "flex",
      "justifyContent": "space-between",
      "margin": "auto",
      "maxWidth": "70%",
    }
  }
>
  <span>
    This is a todo
  </span>
  <span>
    This todo is not done
  </span>
  <span>
    <button
      onClick={[Function]}
    >
       Delete 
    </button>
    <button
      onClick={[Function]}
    >
       Set as done 
    </button>
  </span>
</div>
`;
